contrib/trace_propagators/src/main/java/io/opentelemetry/contrib/trace/propagation/AwsXRayPropagator.java

```diff
@@ -0,0 +1,266 @@
+  }
+
+  @SuppressWarnings("StringSplitter")
+  private static <C> SpanContext getSpanContextFromHeader(C carrier, Getter<C> getter) {
+    String traceHeader = getter.get(carrier, TRACE_HEADER_KEY);
+    if (Strings.isNullOrEmpty(traceHeader)) {
+      return SpanContext.getInvalid();
+    }
+
+    TraceId traceId = TraceId.getInvalid();
+    SpanId spanId = SpanId.getInvalid();
+    TraceFlags traceFlags = TraceFlags.getDefault();
+
+    int pos = 0;
+    while (pos < traceHeader.length()) {
+      int delimiterIndex = traceHeader.indexOf(TRACE_HEADER_DELIMITER, pos);
+      final String part;
+      if (delimiterIndex >= 0) {
+        part = traceHeader.substring(pos, delimiterIndex);
+        pos = delimiterIndex + 1;
+      } else {
+        // Last part.
+        part = traceHeader.substring(pos);
+        pos = traceHeader.length();
+      }
+      String trimmedPart = part.trim();
+      int equalsIndex = trimmedPart.indexOf(KV_DELIMITER);
+      if (equalsIndex < 0) {
+        logger.info(
+            "Error parsing X-Ray trace header. Invalid key value pair: "
+                + part
+                + " Returning INVALID span context.");
+        return SpanContext.getInvalid();
+      }
+
+      String value = trimmedPart.substring(equalsIndex + 1);
+
+      if (trimmedPart.startsWith(TRACE_ID_KEY)) {
+        traceId = parseTraceId(value);
+      } else if (trimmedPart.startsWith(PARENT_ID_KEY)) {
+        spanId = parseSpanId(value);
+      } else if (trimmedPart.startsWith(SAMPLED_FLAG_KEY)) {
+        traceFlags = parseTraceFlag(value);
+      }
+      // TODO: Put the arbitrary TraceHeader keys in OT trace state
+    }
+
+    if (!traceId.isValid()) {
+      logger.info(
+          "Invalid TraceId in X-Ray trace header: '"
+              + TRACE_HEADER_KEY
+              + "' with value "
+              + traceHeader
+              + "'. Returning INVALID span context.");
+      return SpanContext.getInvalid();
+    }
+
+    if (!spanId.isValid()) {
+      logger.info(
+          "Invalid ParentId in X-Ray trace header: '"
+              + TRACE_HEADER_KEY
+              + "' with value "
+              + traceHeader
+              + "'. Returning INVALID span context.");
+      return SpanContext.getInvalid();
+    }
+
+    if (traceFlags == null) {
+      logger.info(
+          "Invalid Sampling flag in X-Ray trace header: '"
+              + TRACE_HEADER_KEY
+              + "' with value "
+              + traceHeader
+              + "'. Returning INVALID span context.");
+      return SpanContext.getInvalid();
+    }
+
+    return SpanContext.createFromRemoteParent(traceId, spanId, traceFlags, TraceState.getDefault());
+  }
+
+  private static TraceId parseTraceId(String xrayTraceId) {
+    if (xrayTraceId.length() != TRACE_ID_LENGTH) {
+      return TraceId.getInvalid();
+    }
+
+    // Check version trace id version
+    if (xrayTraceId.charAt(0) != TRACE_ID_VERSION) {
```

Comment:
If you made the TRACE_ID_VERSION a string, you could avoid the String.valueOf call above, and just use `startsWith` for this call. Might read a little cleaner in both cases.